hk_adc/hk_adc.c

```diff
@@ -0,0 +1,530 @@
+    // By default, ADC is configured to point towards CITI_TEMP as MUX input
+    uint16_t config = HK_ADC_CONFIG_MUX_SINGLE_2 | HK_ADC_CONFIG_FSR_2 | HK_ADC_CONFIG_MODE_SINGLE_SHOT | HK_ADC_CONFIG_DR_1600SPS | HK_ADC_CONFIG_CMODE_TRAD | HK_ADC_CONFIG_CPOL_ACTIVE_LOW | HK_ADC_CONFIG_CLAT_NON_LATCH | HK_ADC_CONFIG_CQUE_DISABLE;
+    uint8_t send_buffer[2] = {config>>8, config};
+
+    // Set the clock for i2c0 instance
+    MSS_I2C_init(&g_mss_i2c0, DUMMY_SERIAL_ADDR, MSS_I2C_PCLK_DIV_256); //10^8/256 = 390,625Hz =~ 390KHz
+
+    // Write to the CONFIG register
+    err = hk_adc_reg_write(HK_ADC_REG_CONFIG, &send_buffer[0]);
+
+    if (err == HK_ADC_NO_ERR)
+    {
+        return err;
+    }
+
+    err = HK_ADC_ERR_INIT_FAILED;
+    return err;
+}
+
+
+
+
+/**
+ * @brief HK ADC read Battery Voltage function
+ *
+ * This function returns the raw battery voltage value as read by the ADC. ADC
+ * measures raw Vout value where Vout = V_Batt/8. V_Batt is found by performing
+ * appropriate conversion on the ground software (CITIROC UI tool) side. Raw ADC
+ * value is returned in the unsigned 16-bit parameter passed to the function as
+ * argument.
+ *
+ * @param batt_volt[in]     pointer to unsigned 16-bit variable to store the result in.
+ * @return HK_ADC_ERR_VOLT_READ_FAILED if any error occurs otherwise HK_ADC_NO_ERR.
+ *
+ */
+int hk_adc_conv_read_volt(uint16_t * batt_volt)
+{
+    uint8_t err = HK_ADC_ERR_VOLT_READ_FAILED;
+
+    uint8_t send_buffer[2] = {0};
+
+    err = hk_adc_reg_read(HK_ADC_REG_CONFIG, &read_value);
+
+    if (err == HK_ADC_NO_ERR)
+    {
+        // Let the rest of the configurations remain intact, just change MUX input config and start single-shot conversion
+        read_value = read_value & 0x8fff;
+        read_value = read_value | HK_ADC_CONFIG_MUX_SINGLE_1;
+
+        send_buffer[0] = (read_value >> 8) | HK_ADC_CONFIG_OS_SINGLE_CONV;
+        send_buffer[1] = read_value;
+
+        // Write changed MUX input back to the CONFIG register
+        err = hk_adc_reg_write(HK_ADC_REG_CONFIG, &send_buffer[0]);
+
+        if (err == HK_ADC_NO_ERR)
+        {
+            //INSERT DELAY FN (1sec), do not use WHILE Loops
+            //Wait until OS bit turns 1
+            do{
+                hk_adc_reg_read(HK_ADC_REG_CONFIG, &read_value);
+                os_bit = read_value >> 15;
+            }while(os_bit == HK_ADC_CONFIG_OS_NOT_READY);
+
+            err = hk_adc_reg_read(HK_ADC_REG_CONVERSION, &read_value);
+            if (err == HK_ADC_NO_ERR)
+            {
+                // Right shift the received result by 4 bits. Conversion value are 12 most significant bits
+                read_value = read_value >> 4;
+                *batt_volt = read_value;
+                return err;
+            }
+        }
+    }
+
+    err = HK_ADC_ERR_VOLT_READ_FAILED;
+    return err;
+}
+
+
+
+
+/**
+ * @brief HK ADC read Battery Current function
+ *
+ * This function returns the raw battery current value as read by the ADC. ADC
```

Comment:
I would rephrase "returns" here, to avoid confusion in the future.

Reply:
Same as above